File: PlanetExploration/PlanetExploration.Core/Models/Planet.cs
namespace PlanetExploration.PlanetExploration.Core.Models
{
    public class Planet
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public PlanetStatus Status { get; set; } = PlanetStatus.EnRoute;
        public string? ImageUrl { get; set; }
        public virtual HumanCaptain? HumanCaptain { get; set; }
        public virtual ICollection<Robot>? Robots { get; set; } = new HashSet<Robot>();
    }

    public enum PlanetStatus
    {
        EnRoute,
        OK,
        NotOK,
    }
}

File: planet-exploration-frontend/src/app/interfaces/planet.ts
import { IHumanCaptain } from "./human-captain";
import { IRobot } from "./robot";

export interface IPlanet {
    id: number;
    name: string;
    description?: string;
    status: number;
    imageUrl: string;
    humanCaptain: IHumanCaptain;
    robots: IRobot[];
}